src/qtcore/qml/QMLBaseObject.js

```diff
@@ -14,6 +14,10 @@ function QMLBaseObject(meta) {
     engine.completedSignals.push(this.Component.completed);
     this.completed = this.Component.completed;
 
+    this.Component.destruction = Signal([]);
+    //engine.destructionSignals.push(this.Component.destruction); would we need this?
```

Comment:
@pavelvasev Are you proposing to pull this in as is (with the comment) and to sort it out later?
Would be fine by me, if the comment is changed to a TODO and a notice about graceful engine shutdown.